packages/forms/src/Components/RichEditor.php

```diff
@@ -820,4 +821,68 @@ public function getFloatingToolbars(): array
     {
         return $this->evaluate($this->floatingToolbars) ?? $this->getDefaultFloatingToolbars();
     }
+
+    public function getLengthValidationRules(): array
+    {
+        $rules = [];
+
+        if (filled($maxLength = $this->getMaxLength())) {
+            $rules[] = function (string $attribute, mixed $value, Closure $fail) use ($maxLength): void {
+                if (blank($value)) {
+                    return;
+                }
+
+                $textLength = Str::length($this->getTipTapEditor()
+                    ->setContent($value)
+                    ->getText());
+
+                if ($textLength > $maxLength) {
+                    $fail(__('validation.max.string', [
+                        'attribute' => str_replace('data.', '', $attribute),
+                        'max' => $maxLength,
+                    ]));
```

Comment:
I think this is the correct way to handle a translatable validation rule so `:attribute` is passed automatically and has the correct label, but please test it first

```suggestion
                    $fail('validation.max.string')->translate([
                        'max' => $maxLength,
                    ]);
```

Reply:
Thanks @danharrin, I just made the changes and it still works as expected 🙏